### DISCUSSION TITLE
[@passport-google-oauth 2.0] GoogleCallbackParameters snake_case outdated

### DISCUSSION THREAD

**Discussion by GabriMagic** [DefinitelyTyped/DefinitelyTyped#58005] [Issues with a @types package]
Hi, I was working with Google authentication when I figured `GoogleCallbackParameters` exists, but when I tried to use it, I found that some properties came in camelCase (all except `id_token`, then I for undefined if I type them:

![image](https://user-images.githubusercontent.com/6409499/148331710-b9f9ffdd-afda-4739-a319-e280c4e8cf72.png)

Is this some outdated bug? Or maybe library inconsistencies? It would be nice to know if someone has noticed the same

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e605162d378f52f5cb8fccb137442320c876bf4d/types/passport-google-oauth20/index.d.ts#L280-L287

I am using this versions:

```
"passport-google-oauth20": "^2.0.0"
"passport": "^0.5.2"
```

Thanks in advance.

[a comment by an automated account is not shown]

**Comment by martin-badin** [comment]
@GabriMagic Please, could you add the URL for documentation?

**Reply by martin-badin** [reply]
PR is ready https://github.com/DefinitelyTyped/DefinitelyTyped/pull/58088 but it will be better to have documentation.

**Reply by GabriMagic** [reply]
I didn not found the exactly documentation, I was debugging the google callback as it is possible and I got this response, every documentation that I found related with the Google Callback use camelCase for instance: (https://medium.com/@vijaysm97/google-oauth20-using-passport-js-abf8928fceed)

![image](https://user-images.githubusercontent.com/6409499/149045875-f1ee16ae-6a9f-4650-820c-c18f5f555183.png)

**Reply by ezintz** [reply]
I am not quite sure, tbh. Google itself seems to still referencing it as `access_token`. https://developers.google.com/identity/protocols/oauth2/javascript-implicit-flow#example

Is it related to passport, passport-google-oauth20 or even Nest.js? Haven't been using it for a long time. Maybe I can give it later a try.

**Reply by GabriMagic** [reply]
As I said I'm not sure about that also, maybe is some dependency in the middle of the process, but I received most of the params in camelCase, that's why I have this issue. I use:

- @nestjs/passport
- passport-google-oauth20

I don't know if something is happening in the middle there.
Thanks for your answer

**Reply by ezintz** [reply]
Even after enabling `passReqToCallback: true,` in my GoogleOAuthStrategy, the `validate` method does not contain any `accessToken`, but also no `access_token`. The signature of the method looks for me

```typescript
public async validate(
  accessToken: string,
  refreshToken: string,
  profile: Profile,
): Promise<UserEntity> {
```

where Profile is either the `Profile` or the request passed to the AuthGuard with respective strategy. The NestJS documentation states the following:

> If a user is found and the credentials are valid, the user is returned so Passport can complete its tasks (e.g., creating the user property on the Request object), and the request handling pipeline can continue. 

Is anyone experiencing issues?
